src/controller.cpp

```diff
@@ -194,13 +194,9 @@ void Controller::activateInternal() {
   // Multihop connections provide a list of servers, starting with the exit
   // node as the first element, and the entry node as the final entry.
   QList<Server> serverList = {server};
-  if (FeatureMultiHop::instance()->isSupported() &&
-      settingsHolder->multihopTunnel()) {
-    ServerData data;
-    Server hop = vpn->randomHop(data);
-    while (serverList.contains(hop)) {
-      hop = vpn->randomHop(data);
-    }
+  if (FeatureMultiHop::instance()->isSupported() && vpn->multihop()) {
```

Comment:
Maybe we can move the `isSupported()` check in the `multihop()` method.

Reply:
I prefer to keep the stuff in the `models` directory to be just plain data containers, and to leave these kinds of conditionals in the root objects of the project. But it's not a big deal to change it either.

Reply:
Actually, it turns out that calling `isSupported()` from within a unit test results in a segmentation fault. And I only realized that after pushing. Oops.